packages/upload-client/src/blob.js

```diff
@@ -303,7 +338,57 @@ export async function add(
     })
   }
 
-  return multihash
+  // Ensure the blob has been accepted
+  const acceptReceipt = await retry(
+    async () => {
+      try {
+        return await getReceipt(nextTasks.accept.task.link(), options)
+      } catch (err) {
+        throw new Error(`failed ${BlobCapabilities.add.can} invocation`, {
```

Comment:
Is this the correct message here?

Reply:
It would be nice to distinguish between not there yet vs an actual error, and quit early from retry if the latter. Typically new code returns `Result` types, which allow us to properly type possible error states. So we might do something like:

```js
const acceptReceipt = await retry(async () => {
  const res = await getReceipt(nextTasks.accept.task.link(), options)
  if (res.error) {
    if (res.error.name === 'ReceiptNotFound') {
      // throw an error that will cause `p-retry` to retry with
      throw new Error('receipt not yet available')
    } else {
      // bail the retry using `import { AbortError } from 'p-retry'`
      throw new AbortError(new Error('failed to fetch blob/accept receipt', { cause: res.error }))
    }
  }
  return res.ok
})
```

Not 100% necessary but I would then consider extracting this as a utility like:

```js
const acceptReceipt = await Receipt.poll(nextTasks.accept.task.link(), options)
```

Reply:
working on this 🙏

Reply:
done